### PR TITLE
Add scan & query filter support to FakeDynamo

### PR DESCRIPTION
**Pull request by chaosgame**
Hello @xiao, 

Please review the following commits I made in branch 'nathan-AttributeValueList'.

97b1fd69e5f1da93af6385cf574b21f3e8aa1d61 (2015-07-27 09:33:08 -0700)
Add scan & query filter support to FakeDynamo

R=@xiao

Test plan: 'New unit test'

### CODE REVIEW

File: lib/ConditionBuilder.js [hunk 1]
```diff
@@ -19,6 +19,47 @@ ConditionBuilder.populateFieldFromConditionBuilderList = function (field, condit
   }
 }
 
+function comparisonOp(operator) {
+
+}
```
Comment (kylehg) [inline lib/ConditionBuilder.js]: :wave:

Comment (chaosgame) [inline lib/ConditionBuilder.js]: Well that's embarrassing.

File: lib/typeUtil.js [hunk 2]
```diff
@@ -90,28 +90,40 @@ function valueToObject(value) {
...
-    }
-    return numArray
+function objectToValue(obj) {
+  switch (objectToType(obj)) {
+    case 'SS':
+      return obj['SS']
+    case 'S':
+      return obj['S']
+    case 'NS':
+      return obj['NS'].map(function (num) { return Number(num) })
```
Comment (kylehg) [inline lib/typeUtil.js]: Why a `map` here but not for `SS`?

**Comment by kylehg** [comment]
LGTM, though someone more familiar with the Dynamite codebase should maybe review too

File: lib/typeUtil.js (hunk 2 above)
Comment (chaosgame) [inline lib/typeUtil.js]: For SS and NS we're given back a list of strings.  SS needs to return a list of strings (already good) and NS needs to return a list of integers (needs the `.map` code).

(Note I just roughly copied the code that was here previously).

**Pull request merged**